library/core/class.oauth2.php

```diff
@@ -220,13 +222,15 @@ public function setRequestAccessTokenParams($params) {
 
 
     /**
-     * Set additional params to be added to the get string in the getProfile request.
+     * Set additional params to be to be merged with the default parameters
+     * in the getProfile request.
+     *
+     * @param array $params Params to add to the get profile request.
      *
-     * @param string $params.
      * @return $this Return this object for chaining purposes.
      */
-    public function setGetProfileParams($params) {
-        $this->getProfileParams = $params;
+    public function setRequestProfileParams($params = []) {
```

Comment:
Why did you default $params to an empty array?
The less default parameters the better. It also doesn't make sense to me in this context because it would not be very useful to call `$obj->setRequestProfileParams();`